huntsman/observatory.py

```diff
@@ -316,6 +316,79 @@ def take_flat_fields(self, camera_names=None, alt=None, az=None,
 
         self.logger.info('Finished flat-fielding.')
 
+    def activate_camera_cooling(self):
+        """
+        Activate camera cooling for all cameras.
+        """
+        self.logger.debug('Activating camera cooling for all cameras.')
+        for cam in self.cameras.values():
+            if cam.is_cooled_camera:
+                cam.cooling_enabled = True
+
+    def deactivate_camera_cooling(self):
+        """
+        Deactivate camera cooling for all cameras.
+        """
+        self.logger.debug('Deactivating camera cooling for all cameras.')
+        for cam in self.cameras.values():
+            if cam.is_cooled_camera:
+                cam.cooling_enabled = False
+
+    def prepare_cameras(self, sleep=60, max_attempts=5, require_all_cameras=False):
+        """
+        Make sure cameras are all cooled and ready.
+
+        Arguments:
+            sleep (float): Time in seconds to sleep between checking readiness. Default 60.
+            max_attempts (int): Maximum number of ready checks. See `require_all_cameras`.
+            require_all_cameras (bool): `True` if all cameras are required to be ready.
+                If `True` and max_attempts is reached, a `PanError` will be raised. If `False`,
+                any camera that has failed to become ready will be dropped from the Observatory.
+        """
+        # Make sure camera cooling is enabled
+        self.activate_camera_cooling()
+
+        # Wait for cameras to be ready
+        n_cameras = len(self.cameras)
+        self.logger.debug('Waiting for cameras to be ready.')
+        for i in range(max_attempts):
```

Comment:
Easier and more reliable to handle the subtraction here:
```suggestion
        for i in range(max_attempts - 1):
```

Reply:
Doesn't this just make the number of checks decrease by 1? Perhaps you mean `for i in range(1, max_attempts+1)`?